write/batcher.go

```diff
@@ -96,7 +96,7 @@ func (b *Batcher) read(ctx context.Context, r io.Reader, lines chan<- []byte, er
 	for scanner.Scan() {
 		// exit early if the context is done
 		select {
-		case lines <- []byte(scanner.Text()):
+		case lines <- scanner.Bytes():
```

Comment:
>  The underlying array may point to data that will be overwritten by a subsequent call to Scan

That sounds scary - are we sure this data is not being overwritten? - i.e. we call `Bytes`, put it on the channel again, and call `Scan` before picking the bytes off the channel.

Reply:
Hmmmm maybe that's why we were using `Text()`... I'll restore it, and add a comment for the next person to see

Reply:
Fixed!